code/modules/admin/verbs/map_export.dm

```diff
@@ -164,56 +188,57 @@ GLOBAL_LIST_INIT(save_file_chars, list(
 	var/layers = FLOOR(log(GLOB.save_file_chars.len, turfs_needed) + 0.999,1)
 
 	//Step 1: Run through the area and generate file data
-	var/list/header_chars = list() //The characters of the header
-	var/list/header_dat = list() //The data of the header, lines up with chars
-	var/header = "" //The actual header in text
-	var/contents = "" //The contents in text (bit at the end)
-	var/index = 1
+	var/list/header_data = list() //holds the data of a header -> to its key
+	var/list/header = list() //The actual header in text
+	var/list/contents = list() //The contents in text (bit at the end)
+	var/key_index = 1 // How many keys we've generated so far
 	for(var/z in 0 to depth)
 		for(var/x in 0 to width)
 			contents += "\n([x + 1],1,[z + 1]) = {\"\n"
 			for(var/y in height to 0 step -1)
 				CHECK_TICK
 				//====Get turfs Data====
-				var/turf/place = locate((minx + x), (miny + y), (minz + z))
+				var/turf/place
 				var/area/location
-				var/list/objects
-				var/area/place_area = get_area(place)
+				var/turf/pull_from = locate((minx + x), (miny + y), (minz + z))
 				//If there is nothing there, save as a noop (For odd shapes)
-				if(!place)
+				if(!pull_from)
```

Comment:
```suggestion
				if(isnull(pull_from))
```

shoulda been like this to start imo- musta missed it the first time

Reply:
killing u/crying